server/controllers/stock/core.js

```diff
@@ -259,3 +266,135 @@ function stockManagementProcess(inventories) {
     return inventory;
   });
 }
+
+/**
+ * @function getStockConsumption
+ *
+ * @description returns the monthly (periodic) stock consumption (CM)
+ *
+ * @param {array} periodIds
+ */
+function getStockConsumption(periodIds) {
+  const sql = `
+    SELECT SUM(s.quantity) AS quantity, BUID(i.uuid) AS uuid, i.text, i.code, d.text
+    FROM stock_consumption s
+    JOIN inventory i ON i.uuid = s.inventory_uuid 
+    JOIN depot d ON d.uuid = s.depot_uuid
+    JOIN period p ON p.id = s.period_id
+    WHERE p.id IN (?) 
+    GROUP BY i.uuid, d.uuid
+  `;
+  return db.exec(sql, [periodIds]);
+}
+
+/**
+ * @function getStockConsumptionAverage
+ *
+ * @description returns average of stock consumption (CMM)
+ *
+ * @param {number} periodId - the base period
+ *
+ * @param {number} numberOfMonths - the number of months for calculating the average
+ */
+function getStockConsumptionAverage(periodId, numberOfMonths) {
+  const baseAvgNumberOfMonths = numberOfMonths || 6;
+
+  const queryPeriodRange = `
+    SELECT id FROM period WHERE id BETWEEN ? AND ?;
+  `;
+
+  const queryPeriodId = periodId ?
+    'SELECT id FROM period WHERE id = ? ' :
+    'SELECT id FROM period WHERE CURRENT_DATE() BETWEEN DATE(start_date) AND DATE(end_date);';
+
+  const queryStockConsumption = `
+    SELECT ROUND(AVG(s.quantity)) AS quantity, BUID(i.uuid) AS uuid, i.text, i.code, BUID(d.uuid) AS depot_uuid, d.text AS depot_text
+    FROM stock_consumption s
+    JOIN inventory i ON i.uuid = s.inventory_uuid 
+    JOIN depot d ON d.uuid = s.depot_uuid
+    JOIN period p ON p.id = s.period_id
+    WHERE p.id IN (?) 
+    GROUP BY i.uuid, d.uuid
+  `;
+
+  return db.one(queryPeriodId, [periodId])
```

Comment:
Are you sure you want `db.one()` here?  It looks like it might return multiple rows.  Double check that this still works the way you want it with multiple consumption events.